### PR TITLE
8306133: Open source few AWT Drag & Drop related tests

### PR DESCRIPTION
**Pull request by DamonGuy**
This PR open sources few AWT DnD related tests

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8306133](https://bugs.openjdk.org/browse/JDK-8306133): Open source few AWT Drag &amp; Drop related tests

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**) ⚠️ Review applies to [90e7447d](https://git.openjdk.org/jdk/pull/13591/files/90e7447dcfe5cbb91092d2cef2c30c88315f5c83)
 * [Prasanta Sadhukhan](https://openjdk.org/census#psadhukhan) (@prsadhuk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13591/head:pull/13591` \
`$ git checkout pull/13591`

Update a local copy of the PR: \
`$ git checkout pull/13591` \
`$ git pull https://git.openjdk.org/jdk.git pull/13591/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13591`

View PR using the GUI difftool: \
`$ git pr show -t 13591`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13591.diff">https://git.openjdk.org/jdk/pull/13591.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13591#issuecomment-1518342751)

### CODE REVIEW

File: test/jdk/java/awt/dnd/NextDropActionTest/NextDropActionTest.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+                Util.doDragDrop(robot, startPoint, endPoint);
+                LOCK.wait(WAIT_TIMEOUT);
+            }
+            if (!secondEnd) {
+                System.err.println("DragSourseListener.dragDropEnd() " +
+                        "was not called, returning");
+                return;
+            }
+        } finally {
+            if (frame != null) {
```
Comment (lawrence-andrew) [inline test/jdk/java/awt/dnd/NextDropActionTest/NextDropActionTest.java:159]: frame is created with in EDT but disposing without EDT ?

File: test/jdk/java/awt/dnd/NoTargetNoDragExitTest/NoTargetNoDragExitTest.java [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+                while (!end) {
+                    LOCK.wait();
+                }
+            }
+        } finally {
+            if (frame != null) {
+                frame.dispose();
+            }
+        }
+
```
Comment (lawrence-andrew) [inline test/jdk/java/awt/dnd/NoTargetNoDragExitTest/NoTargetNoDragExitTest.java:136]: You can remove extra lines

File: test/jdk/java/awt/dnd/NextDropActionTest/NextDropActionTest.java (hunk 1 above)
Comment (prrace) [reply test/jdk/java/awt/dnd/NextDropActionTest/NextDropActionTest.java:159]: There's no harm in doing that but I'm not sure its necessary for the dispose().

**Review by prrace — APPROVED** [review]

File: test/jdk/java/awt/dnd/NextDropActionTest/NextDropActionTest.java (hunk 1 above)
Comment (mrserb) [reply test/jdk/java/awt/dnd/NextDropActionTest/NextDropActionTest.java:159]: it is fine to call dispose here on the main, but the frame possibly can be null here.

Comment (DamonGuy) [reply test/jdk/java/awt/dnd/NextDropActionTest/NextDropActionTest.java:159]: I have updated the tests to all dispose on the EDT just in case.

**Comment by DamonGuy** [comment]
Updated the tests with a frame to have a title rather than nothing or a meaningless title.

**Review by prsadhuk — APPROVED** [review]

File: test/jdk/java/awt/dnd/NextDropActionTest/NextDropActionTest.java (hunk 1 above)
Comment (mrserb) [reply test/jdk/java/awt/dnd/NextDropActionTest/NextDropActionTest.java:159]: The problem is not in the "dispose" but in the next check "frame != null" the frame variable can be null here even if frame is shown on the screen.

**Pull request closed without merging**